src/Extension.php

```diff
@@ -203,4 +209,23 @@ protected function loadCallHandlers(ContainerBuilder $container, $errorReporting
         $definition->addTag(CallExtension::CALL_HANDLER_TAG, ['priority' => 50]);
         $container->setDefinition(CallExtension::CALL_HANDLER_TAG . '.runtime', $definition);
     }
+
+    /**
+     * Check whether the extension should bootstrap or not.
+     * The extension should always bootstrap unless the `-h` or `--help` option is passed.
+     */
+    private function getShouldBootstrap(ContainerBuilder $container): bool
+    {
+        if ($this->shouldBootstrap === null) {
+            if ($container->has('cli.input')) {
+                /** @var ArgvInput $input */
+                $input = $container->get('cli.input');
+                $this->shouldBootstrap = !$input->hasParameterOption(['--help', '-h']);
+            } else {
+                // If the input isn't there for some bizarre reason, just assume we should bootstrap.
+                $this->shouldBootstrap = true;
+            }
+        }
+        return $this->shouldBootstrap;
```

Comment:
```suggestion
        if (!$container->has('cli.input)) {
            return true;
        }
        return !$container->get('cli.input')->hasParameterOption(['--help', '-h']);
```
This method is hardly called, no need to cache the result

Reply:
Is it really worth holding back merging for this? 😅
I'll make the change to avoid ping pong

Reply:
Absolutely!

:-)